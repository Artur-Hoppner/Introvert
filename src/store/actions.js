import Axios from 'axios';
import router from '../router/index';

let url;
if (process.env.NODE_ENV === 'development') {
  url = 'http://localhost:5000/api';
} else {
  url = 'https://incognito-backend.herokuapp.com/api';
}
const api = Axios.create({
  baseURL: url
});
//***********************/
//*** GLOBAL ACTIONS ***/
//*********************/
const actions = {
  async login({ state, commit, dispatch }) {
    api
      .post('/users/login', state.login)
      .then(async res => {
        const token = res.data.token;
        const user = res.data.user;
        Axios.defaults.headers.common['Authorization'] = token;
        localStorage.setItem('token', token);
        await commit('changeUserState', user);
        commit('changeTokenState', token);
        dispatch('statusMessageHandeling', res.data.msg);
        router.push('/');
      })
      .catch(async error => {
        dispatch('statusMessageHandeling', error.response.data.msg);
        console.log(error);
      });
  },
  async getUser({ commit }) {
    Axios.get('https://incognito-backend.herokuapp.com/api/users/profile')
      .then(res => {
        console.log('sending getUser');
        const user = res.data.user;
        console.log(res, 'see response');
        commit('changeUserState', user);
      })
      .catch(error => {
        console.log(error.response);
      });
  },
  async registration({ state, dispatch }) {
    api
      .post('/users/register', state.newUser)
      .then(res => {
        console.log(res.data.msg);
        dispatch('statusMessageHandeling', res.data.msg);
        router.push('/login');
      })
      .catch(error => {
        console.log(error.response);
        dispatch('statusMessageHandeling', error.response.data.msg);
      });
  },

  async commentingevent({ state, dispatch }) {
    console.log(state.user.username);
    const userName = state.user.username;
    const comment = state.commentEvent;
    const eventId = '5f69d7c2568199adb10d910f';
    api
      .post('/events/commentingevent', {
        userName,
        comment,
        eventId
      })
      .then(res => {
        dispatch('statusMessageHandeling', res.data.msg);
        console.log(res, 'response');
      })
      .catch(error => {
        console.log(error.response);
        dispatch('statusMessageHandeling', error.response.data.msg);
      });
  },
  async attendingToEvent({ state, dispatch }, event) {
    const eventId = event._id;
    const userName = state.user.username;
    api
      .post('/events/attending', {
        userName,
        eventId
      })
      .then(res => {
        dispatch('getEvents');
        console.log(res, 'response');
      })
      .catch(error => {
        console.log(error.response);
      });
  },
  async likeEvent({ state, dispatch }, event) {
    const eventId = event._id;
    const userName = state.user.username;
    api
      .post('/events/like', {
        userName,
        eventId
      })
      .then(res => {
        console.log(res, 'response');
        dispatch('getEvents');
      })
      .catch(error => {
        console.log(error.response);
      });
  },
  async getEvents({ commit }) {
    api
      .get('/events/all')
      .then(async res => {
        const events = res.data.events[0];
        commit('allEvents', events);
      })
      .catch(error => {
        console.log(error.response);
      });
  },
  async newEvent({ state, dispatch }) {
    api
      .post('/events/register', state.newEvent)
      .then(res => {
        console.log(res, 'created event');
        dispatch('getEvents');
        dispatch('statusMessageHandeling', res.data.msg);
        router.push('/');
      })
      .catch(async error => {
        dispatch('statusMessageHandeling', error.response.data.msg);
      });
  },
  async logOut({ commit }) {
    const token = localStorage.removeItem('token');
    delete Axios.defaults.headers.common['Authorization'];
    router.push('/login');
    const user = '';
    await commit('changeTokenState', token);
    commit('changeUserState', user);
  },
  async statusMessageHandeling({ commit }, value) {
    await commit('statusMessage', value);
    const reset = null;
    setTimeout(() => {
      commit('statusMessage', reset);
    }, 2400);
  },
  filterEvent({ commit }) {
    commit('filterEventMutation');
  },
  resetFilter({ commit }) {
    commit('resetFilter');
  }
};

export default actions;
