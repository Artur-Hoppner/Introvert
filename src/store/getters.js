//***********************/
//*** GLOBAL GETTERS ***/
//*********************/
import { getField } from 'vuex-map-fields';

const getters = {
  getField,
  isLoggedIn: state => state.token,
  user: state => state.user,
  statusMessage: state => state.statusMessage
};

export default getters;
