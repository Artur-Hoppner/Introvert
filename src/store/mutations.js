import { updateField } from 'vuex-map-fields';

//*************************/
//*** GLOBAL MUTATIONS ***/
//***********************/

const mutations = {
  updateField,
  allEvents(state, events) {
    state.allEvents = events;
  },
  changeTokenState(state, token) {
    state.token = token;
    state.errorMessage = null;
    console.log(state.token, 'Change token state');
    console.log(localStorage.getItem('token'), 'show localstorage');
  },

  changeUserState(state, user) {
    state.user = user;
    state.newEvent.createdBy = user.username;
  },
  statusMessage(state, response) {
    state.statusMessage = response;
  }
};

export default mutations;
