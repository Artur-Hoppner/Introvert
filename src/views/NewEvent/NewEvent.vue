<template>
  <div id="app">
    <h1>New Event</h1>
    <v-select
      v-model="typeOfEvent"
      :items="[
        'Musical',
        'Theatre',
        'Opera',
        'Lecture',
        'Festival',
        'Dance',
        'Other'
      ]"
      label="Standard"
    ></v-select>

    <input type="text" v-model="name" placeholder="Event Name" />
    <input type="text" v-model="place" placeholder="Location" />
    <input type="date" v-model="date" />
    <div>
      <button @click="newEvent">Create new event</button>
      <button @click="attendingToEvent">attendingToEvent</button>
      <button @click="likeEvent">like Event</button>
      <button @click="commentingevent">Commenting Event</button>
    </div>
  </div>
</template>

<script>
import 'vue-select/dist/vue-select.css';

import { mapActions } from 'vuex';
import { mapFields } from 'vuex-map-fields';
export default {
  name: 'NewEvent',
  computed: {
    ...mapFields([
      'newEvent.name',
      'newEvent.place',
      'newEvent.date',
      'newEvent.typeOfEvent'
    ]),
    storvalue() {
      return this.$store.state;
    }
  },
  methods: {
    ...mapActions([
      'newEvent',
      'getEvents',
      'attendingToEvent',
      'likeEvent',
      'commentingevent'
    ])
  }
};
</script>

<style lang="scss"></style>
