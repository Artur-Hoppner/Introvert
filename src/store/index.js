import Vue from 'vue';
import Vuex from 'vuex';
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex);
//*********************/
//*** GLOBAL STATE ***/
//*******************/

export default new Vuex.Store({
  state: {
    newUser: {
      name: '',
      username: '',
      email: '',
      password: '',
      verifypassword: ''
    },
    login: {
      username: '',
      password: ''
    },
    allEvents: '',
    newEvent: {
      name: '',
      place: '',
      description: '',
      date: '',
      typeOfEvent: '',
      createdBy: ''
    },
    commentEvent: 'testar kommentar p√• event',
    token: localStorage.getItem('token') || '',
    user: '',
    statusMessage: null,
    filterEventTag: '',
    filterEvents: ''
  },
  actions: actions,
  mutations: mutations,
  getters: getters
});
