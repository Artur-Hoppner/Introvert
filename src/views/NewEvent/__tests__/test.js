// Utilities
import { mount, createLocalVue } from '@vue/test-utils';
import { getField, updateField } from 'vuex-map-fields';
import store from '@/store/index.js';
// Libraries
import Vue from 'vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
// Components
import NewEvent from '@/views/NewEvent/NewEvent.vue';

const localVue = createLocalVue();
localVue.use(Vuex, VueRouter);

describe('/newEvent', () => {
  let vuetify;

  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });

  test('Test if newEvent is rendered', () => {
    // Arrange;
    const wrapper = mount(NewEvent, {
      localVue,
      vuetify,
      store
    });
    // Assert
    expect(wrapper.element).toMatchSnapshot();
  });

  test('It should activate newEvent on click', () => {
    const actions = {
      newEvent: jest.fn()
    };
    const store = new Vuex.Store({
      state: {
        newEvent: {
          name: '',
          place: '',
          description: '',
          date: '',
          typeOfEvent: '',
          createdBy: ''
        }
      },
      actions,
      mutations: { updateField },
      getters: { getField }
    });

    // Arrange;
    const wrapper = mount(NewEvent, {
      localVue,
      vuetify,
      store
    });

    const button = wrapper.find('.v-btn');
    wrapper.vm.$on('newEvent');
    expect(actions.newEvent).toHaveBeenCalledTimes(0);
    button.trigger('click');
    expect(actions.newEvent).toHaveBeenCalledTimes(1);
  });
  test('It should pass data to state.newEvent', async () => {
    // Arrange;
    const store = new Vuex.Store({
      state: {
        newEvent: {
          name: '',
          place: '',
          description: '',
          date: '',
          typeOfEvent: '',
          createdBy: ''
        }
      },
      mutations: { updateField },
      getters: { getField }
    });
    const wrapper = mount(NewEvent, {
      localVue,
      vuetify,
      store
    });
    const inputName = wrapper.find('#inputName');
    const inputPlace = wrapper.find('#inputPlace');
    const inputDescription = wrapper.find('#inputDescription');

    inputName.setValue('Phanton of the opera');
    inputPlace.setValue('Stockholm');
    inputDescription.setValue('Its always cool to have a cape');
    await wrapper.vm.$nextTick();

    expect(store.state.newEvent.name).toBe(inputName.element.value);
    expect(store.state.newEvent.place).toBe(inputPlace.element.value);
    expect(store.state.newEvent.description).toBe(
      inputDescription.element.value
    );
  });
});
