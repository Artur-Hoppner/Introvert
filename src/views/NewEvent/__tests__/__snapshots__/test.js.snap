// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/newEvent Test if newEvent is rendered 1`] = `
<div
  class="container container"
>
  <div
    class="layout"
  >
    <div
      class="flex login-form text-xs-center"
    >
      <div
        class="display-1 mb-3 "
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate material-icons theme--light"
          style="font-size: 40px;"
        >
          create
        </i>
         Create your event newEvent
      
      </div>
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div>
            Event information:
          </div>
           
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input theme--light v-text-field v-select"
            >
              <div
                class="v-input__control"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-owns="list-7"
                  class="v-input__slot"
                  role="button"
                >
                  <div
                    class="v-select__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputTypeOfEvent"
                      style="left: 0px; position: absolute;"
                    >
                      Type of event
                    </label>
                    <div
                      class="v-select__selections"
                    >
                      <input
                        aria-readonly="false"
                        autocomplete="off"
                        id="inputTypeOfEvent"
                        readonly="readonly"
                        type="text"
                      />
                    </div>
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon notranslate mdi mdi-menu-down theme--light"
                        />
                      </div>
                    </div>
                    <input
                      type="hidden"
                    />
                  </div>
                  <div
                    class="v-menu"
                  >
                    <!---->
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    person
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputName"
                      style="left: 0px; position: absolute;"
                    >
                      Name
                    </label>
                    <input
                      id="inputName"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    room
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputPlace"
                      style="left: 0px; position: absolute;"
                    >
                      Place
                    </label>
                    <input
                      id="inputPlace"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    info
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputDescription"
                      style="left: 0px; position: absolute;"
                    >
                      description
                    </label>
                    <input
                      id="inputDescription"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="row"
            >
              <div
                class="col-sm-6 col-md-4 col-12"
              >
                <div
                  class="v-menu"
                >
                  <div
                    class="v-input v-input--is-readonly theme--light v-text-field"
                  >
                    <div
                      class="v-input__prepend-outer"
                    >
                      <div
                        class="v-input__icon v-input__icon--prepend"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon notranslate material-icons theme--light"
                        >
                          event
                        </i>
                      </div>
                    </div>
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label theme--light"
                            for="input-25"
                            style="left: 0px; position: absolute;"
                          >
                            Picker in menu
                          </label>
                          <input
                            aria-expanded="false"
                            aria-haspopup="true"
                            id="input-25"
                            readonly="readonly"
                            role="button"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <transition-group-stub
                            class="v-messages__wrapper"
                            name="message-transition"
                            tag="div"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <!---->
                </div>
              </div>
            </div>
             
            <button
              class="v-btn v-btn--contained theme--light v-size--default"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                Create new event
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
